feat: SMS and email OTP texts

eliobischof opened the pull request:
## Reviewable changes

### Console
- [ ] Makes the two message texts for SMS and email OTP verification customizable via API and console.
- [ ] Omits the buttonText property in the console for both the new SMS OTP verification and the existing phone verification texts.
- [ ] Fixes a console bug for resetting instance level message texts.

### API
- [ ] Manage SMS OTP and Email OTP via API
- [ ] Omits the buttonText property in the API for the new SMS OTP verification text.

### Definition of Ready

- [x] I am happy with the code
- [x] Short description of the feature/issue is added in the pr description
- [x] PR is linked to the corresponding user story
- [x] Acceptance criteria are met
- [x] All open todos and follow ups are defined in a new ticket and justified
- [x] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [x] No debug or dead code
- [x] My code has no repetitions
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [x] Documentation/examples are up-to-date
- [x] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.

livio-a commented on proto/zitadel/admin.proto:
```diff
@@ -2953,7 +2953,7 @@ service AdminService {
 
         option (zitadel.v1.auth_option) = {
-            permission: "iam.policy.delete"
+            permission: "iam.policy.delete" // TODO: iam.policy.write on all reset text endpoints?
```
?

eliobischof replied:
Why is it not iam.policy.write?

livio-a commented on internal/notification/static/i18n/zh.yaml:
```diff
@@ -26,13 +26,19 @@ VerifyPhone:
   Subject: 域名所有权验证
   Greeting: 你好 {{.DisplayName}},
   Text: 域 {{.Domain}} 已被组织使用。您当前的用户 {{.Username}} 不属于此组织。因此，您必须在登录时更改您的电子邮件。我们为此登录创建了一个临时用户名 ({{.TempUsername}})。
-  ButtonText: 登录
```
why is that removed? and where is the sms text?

eliobischof replied:
Can we render buttons in SMS?

eliobischof replied:
LOL, idk

eliobischof replied:
readded these translations

livio-a commented on internal/notification/static/i18n/de.yaml:
```diff
@@ -26,13 +26,25 @@ VerifyPhone:
   Subject: Domain wurde beansprucht
   Greeting: Hallo {{.DisplayName}},
   Text: Die Domain {{.Domain}} wurde von einer Organisation beansprucht. Dein derzeitiger User {{.Username}} ist nicht Teil dieser Organisation. Daher musst du beim nächsten Login eine neue Email hinterlegen. Für diesen Login haben wir dir einen temporären Usernamen ({{.TempUsername}}) erstellt.
-  ButtonText: Login
```
why is this removed?

eliobischof replied:
fixed

livio-a commented on internal/i18n/i18n.go:
```diff
@@ -13,7 +13,7 @@ import (
 	"github.com/zitadel/logging"
 	"golang.org/x/text/language"
-	"sigs.k8s.io/yaml"
+	"sigs.k8s.io/yaml" // TODO: not gopkg.in/yaml.v3?
```
?

eliobischof replied:
Isit intended to use the k8s yaml library?

livio-a reviewed (changes requested):
i only checked the backend, but there are a lot of TODOs
please also note that (in preparation of making them not required anymore) we changed the `{{.FirstName}} {{.LastName}}` combination to {{.DisplayName}}

eliobischof commented:
> please also note that (in preparation of making them not required anymore) we changed the `{{.FirstName}} {{.LastName}}` combination to {{.DisplayName}}

I fixed the translations to use DisplayName now.

> i only checked the backend, but there are a lot of TODOs 

Yes, these are questions to you in person :smile: Can you answer them? I'll then create follow-ups or delete the TODOs.

peintnermax commented:
@eliobischof there are some translations missing for:
```
 "VSO": "Verify SMS OTP",
 "VEO": "Verify Email OTP",
```
in all but en.json. The rest looks good IMO

eliobischof commented:
> @eliobischof there are some translations missing for:
> 
> ```
>  "VSO": "Verify SMS OTP",
>  "VEO": "Verify Email OTP",
> ```
> 
> in all but en.json. The rest looks good IMO

done, thanks

eliobischof commented:
@livio-a I removed all TODO comments and the pointless sms text properties. Can you review again, plz?

eliobischof commented on internal/notification/static/i18n/en.yaml:
```diff
@@ -26,6 +26,15 @@ VerifyPhone:
+  Text: Please use the "Authenticate" button or copy the one-time password {{.OTP}} and paste it to to the authentication screen in order to authenticate at ZITADEL within the next five minutes.
+  ButtonText: Authenticate
+VerifySMSOTP:
+  Text: Please visit {{ .VerifyURL }} or copy the one-time password {{.OTP}} and paste it to to the authentication screen in order to authenticate at ZITADEL within the next five minutes.
```
@livio-a is this default text correct?

livio-a replied:
LGTM

livio-a commented:
if checked out the current state and run it locally
LGTM, but e.g. at `Verify SMS OTP` there's a `details` and should we hide the texts, which cannot be set?
![image](https://github.com/zitadel/zitadel/assets/9405495/c328b1eb-c983-40f7-a9df-39392f317286)

cc @peintnermax

eliobischof commented:
> if checked out the current state and run it locally
> LGTM, but e.g. at `Verify SMS OTP` there's a `details` and should we hide the texts, which cannot be set?
> ![image](https://github.com/zitadel/zitadel/assets/9405495/c328b1eb-c983-40f7-a9df-39392f317286)
> 
> cc @peintnermax 

Did you rebuild the console?

livio-a commented:
> 
> Did you rebuild the console?

i used make and probably missed a step to embed it into my local binary. just retried it and now it works, sorry.
